tests/tests/compiler/function/unit_parameter_fail.leo

```diff
@@ -0,0 +1,8 @@
+/*
+namespace = "Compile"
+expectation = "Fail"
+*/
+
+program test.aleo {    
+    transition main(x: ()) {}
```

Comment:
Consider augmenting this test with unit types inside an array or struct, as both the parameter and output.

Reply:
Do we test the case `let x: () = foo()` where `foo` returns the unit type?